hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/SubmissionManager.java

```diff
@@ -130,8 +133,12 @@ public void submit(@NonNull final TransactionBody txBody, @NonNull final Bytes t
         if (txBody.hasUncheckedSubmit()) {
             // We do NOT allow this call in production!
             // check profile dynamically, this way we allow profile overriding in Hapi tests
-            final var hederaConfig = configProvider.getConfiguration().getConfigData(HederaConfig.class);
-            if (hederaConfig.activeProfile() == Profile.PROD) {
+            final var configuration = configProvider.getConfiguration();
+            final var hederaConfig = configuration.getConfigData(HederaConfig.class);
+            final var ledgerConfig = configuration.getConfigData(LedgerConfig.class);
+            if (hederaConfig.activeProfile() == Profile.PROD
+                    || MAIN_NET_LEDGER_ID.equals(ledgerConfig.id())
+                    || TEST_NET_LEDGER_ID.equals(ledgerConfig.id())) {
```

Comment:
We should not allow in previewnet also right ?

Reply:
Added a check for previewnet